lib/document.js

```diff
@@ -1783,18 +1783,22 @@ Document.prototype.$inc = function $inc(path, val) {
     return this;
   }
 
+  const currentValue = this.$__getValue(path) || 0;
+  let valToSet = currentValue + val;
```

Comment:
is this correct to not have `castNumber(val)` too? (like in the try block)

Reply:
You're right. My assumption was that setting the invalid value if casting fails, we should still set the value. But that isn't correct. For example, below script still prints 42:

```javascript
'use strict';

const mongoose = require('mongoose');

run().catch(err => console.log(err));

async function run() {
  await mongoose.connect('mongodb://127.0.0.1/mongoose_test');
  const Test = mongoose.model('Test', mongoose.Schema({ num: Number }));

  const doc = new Test({ num: 42 });
  doc.num = 'abc';
  console.log(doc.num);
}
```

Thanks for calling that out. Can you re-review please?